internal/s3select/select.go

```diff
@@ -577,6 +577,9 @@ OuterLoop:
 
 // Close - closes opened S3 object.
 func (s3Select *S3Select) Close() error {
+	if s3Select.recordReader == nil {
+		return nil
```

Comment:
Not sure, maybe errors.New("closing a not initialized s3select")

Reply:
No, it is not an error. This can happen as per the repro. An error is already being returned.